### DISCUSSION TITLE
Add examples of non-`get_` getters from standard library to the C-GETTER guideline.

### DISCUSSION THREAD

**Discussion by ely-uf** [rust-lang-nursery/api-guidelines#222] [Questions]
The beginning of the C-GETTER guideline states:
> With a few exceptions, the `get_` prefix is not used for getters in Rust code.

However, all the guideline examples from the standard library have the `get_` prefix:
>  `std::io::Cursor::get_mut`
    `std::ptr::Unique::get_mut`
    `std::sync::PoisonError::get_mut`
    `std::sync::atomic::AtomicBool::get_mut`
    `std::collections::hash_map::OccupiedEntry::get_mut`
    `<[T]>::get_unchecked`

representing only the following rule of the guideline:
> The `get` naming is used only when there is a single and obvious thing that could reasonably be gotten by a getter. For example `Cell::get` accesses the content of a `Cell`.

Suggested examples of non-`get_` getters from the standard library:
- [`std::vec::Vec::len`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.len)
- [`std::thread::JoinHandle::thread`](https://doc.rust-lang.org/std/thread/struct.JoinHandle.html#method.thread)
- [`std::panic::Location::file`](https://doc.rust-lang.org/std/panic/struct.Location.html#method.file)

**Comment by KodrAus** [comment]
I think the distinction there is between getters as accessors for one of many fields such as `MyStruct::get_field1_mut` and a single getter for the inner value such as `Cursor::get_mut`. In the former case, we'd prefer `MyStruct::field1_mut`.
